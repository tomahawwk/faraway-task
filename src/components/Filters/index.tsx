import {useEffect} from 'react';
import {fetchCharacters} from '../../store/reducers/ActionCreators';
import Search from '../Search';
import {useFilters} from 'hooks/useFilters';
import {useAppDispatch} from 'store/hooks/redux';

const Filters = () => {
  const {params, handleSearch} = useFilters();
  const dispatch = useAppDispatch();

  useEffect(() => {
    if (!!Object.keys(params).length) dispatch(fetchCharacters(params));
  }, [params]);

  return (
    <div className="w-full flex gap-md z-20 items-end">
      <div className="animation-fade-y animation-delay-3 w-full">
        <Search
          onChange={e => {
            handleSearch(e.target.value);
          }}
          value={''}
        />
      </div>
    </div>
  );
};

export default Filters;
