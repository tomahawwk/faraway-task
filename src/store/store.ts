import {combineReducers, configureStore} from '@reduxjs/toolkit';
import charactersReducer from './reducers/CharactersSlice';
import characterReducer from './reducers/CharacterSlice';
import filtersReducer from './reducers/FiltersSlice';

const rootReducer = combineReducers({
  charactersReducer,
  characterReducer,
  filtersReducer,
});

export const setupStore = () => {
  return configureStore({
    reducer: rootReducer,
    middleware: getDefaultMiddleware =>
      getDefaultMiddleware({
        serializableCheck: false,
      }),
  });
};

export type RootState = ReturnType<typeof rootReducer>;
export type AppStore = ReturnType<typeof setupStore>;
export type AppDispatch = AppStore['dispatch'];
