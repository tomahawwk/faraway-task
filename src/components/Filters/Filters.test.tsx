import {render, screen} from '@testing-library/react';

import Filters from '.';
import {setupStore} from 'store/store';
import {Provider} from 'react-redux';

const store = setupStore();

describe('Filters component', () => {
  test('renders Filters component', () => {
    render(
      <Provider store={store}>
        <Filters />
      </Provider>,
    );
    const titleElement = screen.getByText('Search characters');
    expect(titleElement).toBeInTheDocument();
  });

  test('Filters snapshot', () => {
    const filters = render(
      <Provider store={store}>
        <Filters />
      </Provider>,
    );

    expect(filters).toMatchSnapshot();
  });
});
