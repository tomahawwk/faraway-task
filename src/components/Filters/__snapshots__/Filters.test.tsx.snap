// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filters component Filters snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="w-full flex gap-md z-20 items-end"
      >
        <div
          class="animation-fade-y animation-delay-3 w-full"
        >
          <div
            class="search w-full"
          >
            <div
              class="relative w-full min-w-[200px] h-11"
            >
              <input
                class="peer w-full h-full bg-transparent text-blue-gray-700 font-sans font-normal outline outline-0 focus:outline-0 disabled:bg-blue-gray-50 disabled:border-0 disabled:cursor-not-allowed transition-all border-b placeholder-shown:border-blue-gray-200 placeholder:opacity-0 focus:placeholder:opacity-100 text-sm pt-4 pb-1.5 !text-white border-white focus:border-white"
                placeholder=" "
              />
              <label
                class="flex w-full h-full select-none pointer-events-none absolute left-0 font-normal !overflow-visible truncate peer-placeholder-shown:text-blue-gray-500 leading-tight peer-focus:leading-tight peer-disabled:text-transparent peer-disabled:peer-placeholder-shown:text-blue-gray-500 transition-all -top-1.5 peer-placeholder-shown:text-sm text-[11px] peer-focus:text-[11px] after:content[''] after:block after:w-full after:absolute after:-bottom-1.5 left-0 after:border-b-2 after:scale-x-0 peer-focus:after:scale-x-100 after:transition-transform after:duration-300 peer-placeholder-shown:leading-[4.25] !text-white peer-focus:text-white after:border-white peer-focus:after:border-white"
              >
                Search characters
                 
              </label>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="w-full flex gap-md z-20 items-end"
    >
      <div
        class="animation-fade-y animation-delay-3 w-full"
      >
        <div
          class="search w-full"
        >
          <div
            class="relative w-full min-w-[200px] h-11"
          >
            <input
              class="peer w-full h-full bg-transparent text-blue-gray-700 font-sans font-normal outline outline-0 focus:outline-0 disabled:bg-blue-gray-50 disabled:border-0 disabled:cursor-not-allowed transition-all border-b placeholder-shown:border-blue-gray-200 placeholder:opacity-0 focus:placeholder:opacity-100 text-sm pt-4 pb-1.5 !text-white border-white focus:border-white"
              placeholder=" "
            />
            <label
              class="flex w-full h-full select-none pointer-events-none absolute left-0 font-normal !overflow-visible truncate peer-placeholder-shown:text-blue-gray-500 leading-tight peer-focus:leading-tight peer-disabled:text-transparent peer-disabled:peer-placeholder-shown:text-blue-gray-500 transition-all -top-1.5 peer-placeholder-shown:text-sm text-[11px] peer-focus:text-[11px] after:content[''] after:block after:w-full after:absolute after:-bottom-1.5 left-0 after:border-b-2 after:scale-x-0 peer-focus:after:scale-x-100 after:transition-transform after:duration-300 peer-placeholder-shown:leading-[4.25] !text-white peer-focus:text-white after:border-white peer-focus:after:border-white"
            >
              Search characters
               
            </label>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
